### PR TITLE
[2018-06] Crash Reporter V2

### PR DESCRIPTION
**Pull request by alexanderkyte**
Let's hold off merging this until we have commits from

https://github.com/mono/mono/pull/11133

and

#11142

to cherry-pick onto this. These requested changes overlap with the below commits.

Backport https://github.com/mono/mono/pull/10532

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -5749,6 +5749,10 @@ echo "
 	zlib:            $zlib_msg
 	BTLS:            $enable_btls$btls_platform_string
 	jemalloc:        $with_jemalloc (always use: $with_jemalloc_always)
+<<<<<<< HEAD
```
Comment (jaykrell) [inline configure.ac]: merge conflict marker

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -5921,6 +5921,97 @@ ves_icall_Mono_Runtime_SendMicrosoftTelemetry (char *payload, guint64 portable_h
...
+ves_icall_Mono_Runtime_DumpTelemetry (char *payload, guint64 portable_hash, guint64 unportable_hash, MonoError *error)
+{
+#ifdef TARGET_OSX
+	MonoStackHash hashes;
+	memset (&hashes, 0, sizeof (MonoStackHash));
+	hashes.offset_free_hash = portable_hash;
+	hashes.offset_rich_hash = unportable_hash;
+	mono_crash_dump (payload, &hashes);
+#else
+	return;
```
Comment (jaykrell) [inline mono/metadata/icall.c]: not needed

**Comment by luhenry** [comment]
@alexanderkyte both https://github.com/mono/mono/pull/11133 and https://github.com/mono/mono/pull/11142 have now been merged.

**Comment by alexanderkyte** [comment]
@monojenkins build deb with monolite

**Comment by alexanderkyte** [comment]
Despite including the enclosing namespace, 

```

Test/System/ExceptionTest.cs(417,58): error CS0103: The name `BindingFlags' does not exist in the current context
Test/System/ExceptionTest.cs(480,56): error CS0103: The name `BindingFlags' does not exist in the current context
Test/System/ExceptionTest.cs(495,55): error CS0103: The name `BindingFlags' does not exist in the current context

``` 

when using `mcs`, so I used the fully qualified name here.

**Comment by alexanderkyte** [comment]
Looks like I need to disable the added tests on non-osx platforms.

**Comment by alexanderkyte** [comment]
That disables exactly the things that failed before. Should be green. Would like to merge soon.

**Comment by luhenry** [comment]
@monojenkins squash

**Comment by alexanderkyte** [comment]
```20:28:12 Tests run: 10722, Passed: 10624, Errors: 0, Failures: 0, Inconclusive: 0
20:28:12   Not run: 98, Invalid: 0, Ignored: 98, Skipped: 0
20:28:12 Elapsed time: 00:23:22.8080000
20:28:19 ``` 

on the interpreter 

```
20:28:29 
20:28:36 ............
20:28:36 Tests run: 10, Passed: 10, Errors: 0, Failures: 0, Inconclusive: 0
20:28:36   Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0
20:28:36 Elapsed time: 00:00:07.0720000
20:28:36 make: Leaving directory '/home/builder/jenkins/workspace/test-mono-pull-request-interpreter-armhf/mcs/class/corlib'
20:28:39 *** end(1474): corlib: Passed
```

**Pull request merged**